aten/src/ATen/TensorIterator.cpp

```diff
@@ -820,6 +820,27 @@ void TensorIteratorBase::build_binary_op(const Tensor& out, const Tensor& a, con
     .enforce_safe_casting_to_output(true));
 }
 
+void TensorIteratorBase::build_unary_float_op(const Tensor& out, const Tensor& a) {
+  build(TensorIteratorConfig()
+      .set_check_mem_overlap(true)
+      .add_output(out)
+      .add_input(a)
+      .promote_inputs_to_common_dtype(true)
+      .cast_common_dtype_to_outputs(true)
+      .enforce_safe_casting_to_output(true)
+      .promote_integer_inputs_to_float(true));
+}
+
+void TensorIteratorBase::build_unary_op(const Tensor& out, const Tensor& a) {
```

Comment:
Just to cut down on code duplication- could you update `TensorIterator::unary_op` and `unary_float_op` to call these? You can do something similar to what `TensorIterator::binary_op` currently does.

Reply:
Done!